src/Phoenix.vue

```diff
@@ -39,17 +51,35 @@ export default {
   },
   computed: {
     ...mapState(['route', 'user']),
-    ...mapGetters(['configuration', 'activeNotifications', 'activeMessages', 'capabilities']),
+    ...mapGetters(['configuration', 'activeNotifications', 'activeMessages', 'capabilities', 'apps']),
     $_applicationsList () {
-      return this.configuration.applications
+      const list = []
+
+      // Get extensions manually added into config
+      list.push(this.configuration.applications)
+
+      // Get extensions which have at least one nav item
+      const navItems = this.$root.navItems
+      for (const item in navItems) {
+        list.push(this.apps[item])
```

Comment:
confusing: does `navItems` contain app ids somewhere ?
from reading the code it seems `navItems` keys are the app ids and you're taking them from there.
if yes I suggest renaming `item` to `appId` to make this clearer

Reply:
NavItems for each extension are stored under extension id. I changed the name. You're right it wasn't clear.